packages/terra-badge/src/Badge.jsx

```diff
@@ -57,8 +65,9 @@ const Badge = ({
     customProps.className,
   );
 
-  const textContent = text ? <span className={styles.text}>{text}</span> : null;
-  const content = isReversed ? [textContent, icon, customProps.children] : [icon, textContent, customProps.children];
+  const textContent = text ? <span className={styles.text} aria-label={visuallyHiddenText}>{text}</span> : null;
```

Comment:
Let's remove the aria-label. This results in the visually hidden text being read twice on VoiceOver on desktop, it also prevents VoiceOver on desktop from reading the `{text}` content as aria-label is read instead.

VoiceOver on iOS does not read the aria-label so the use of `<VisuallyHiddenText />` component is needed.